core/tests/protractor_utils/AdminPage.js

```diff
@@ -373,6 +373,10 @@ var AdminPage = function() {
 
   this.expectUsernamesToMatch = async function(expectedUsernamesArray) {
     var foundUsersArray = [];
+    if (expectedUsernamesArray.length !== 0) {
+      await waitFor.visibilityOf(element(
+        by.css('.protractor-test-roles-result-rows')));
+    }
```

Comment:
@DubeySandeep 

We currently don't have an expectation where the length of `expectedUsernamesArray` is 0. The `if` block is, in fact, not needed (I added it just in case).

If someone in the future does add an expectation where the length of `expectedUsernamesArray` is 0, they can write a separate `expectNoUsernamesToMatch()` function.

I think we need to get this PR merged ASAP since the `adminPage` suite only passes 20% of the time right now.

Reply:
Lets merge this PR because of the following reasons:

1. It will fix major flakiness issue 
2. We are refactoring the roles tab and changing the complete UI through #12916, so expectedUsernamesArray is going to be removed in the same PR!